cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/LedgerEvents/Gov/PredefinedAbstainDRep.hs

```diff
@@ -0,0 +1,362 @@
+      era = toCardanoEra sbe
+      cEra = AnyCardanoEra era
+      fastTestnetOptions = cardanoDefaultTestnetOptions
+        { cardanoEpochLength = 100
+        , cardanoNodeEra = cEra
+        , cardanoNumDReps = 3
+        }
+
+  testnetRuntime@TestnetRuntime
+    { testnetMagic
+    , poolNodes
+    , wallets=wallet0:wallet1:wallet2:_
+    , configurationFile
+    }
+    <- cardanoTestnetDefault fastTestnetOptions conf
+
+  poolNode1 <- H.headM poolNodes
+  poolSprocket1 <- H.noteShow $ nodeSprocket $ poolRuntime poolNode1
+  execConfig <- H.mkExecConfig tempBaseAbsPath poolSprocket1 testnetMagic
+
+  let socketName' = IO.sprocketName poolSprocket1
+      socketBase = IO.sprocketBase poolSprocket1 -- /tmp
+      socketPath = socketBase </> socketName'
+
+  epochStateView <- getEpochStateView (File configurationFile) (File socketPath)
+
+  startLedgerNewEpochStateLogging testnetRuntime tempAbsPath'
+
+  H.note_ $ "Sprocket: " <> show poolSprocket1
+  H.note_ $ "Abs path: " <> tempAbsBasePath'
+  H.note_ $ "Socketpath: " <> socketPath
+  H.note_ $ "Foldblocks config file: " <> configurationFile
+
+  gov <- H.createDirectoryIfMissing $ work </> "governance"
+
+  initialDesiredNumberOfPools <- getDesiredPoolNumberValue epochStateView ceo
+
+  let newNumberOfDesiredPools = initialDesiredNumberOfPools + 1
+
+  -- Do some proposal and vote yes with the first DRep only
+  -- and assert that proposal does NOT pass.
+  void $ desiredPoolNumberProposalTest execConfig epochStateView configurationFile socketPath ceo gov "firstProposal"
+                                       wallet0 Nothing [(1, "yes")] newNumberOfDesiredPools 3 (Just initialDesiredNumberOfPools) 10
+
+  -- Take the last two stake delegators and delegate them to "Abstain".
+  delegateToAlwaysAbstain execConfig epochStateView configurationFile socketPath sbe gov "delegateToAbstain1"
+                          wallet1 (defaultDelegatorStakeKeyPair 2)
+  delegateToAlwaysAbstain execConfig epochStateView configurationFile socketPath sbe gov "delegateToAbstain2"
+                          wallet2 (defaultDelegatorStakeKeyPair 3)
+
+  -- Do some other proposal and vote yes with first DRep only
+  -- and assert the new proposal passes now.
+  let newNumberOfDesiredPools2 = newNumberOfDesiredPools + 1
+  void $ desiredPoolNumberProposalTest execConfig epochStateView configurationFile socketPath ceo gov "secondProposal"
+                                       wallet0 Nothing [(1, "yes")] newNumberOfDesiredPools2 0 (Just newNumberOfDesiredPools2) 10
+
+delegateToAlwaysAbstain
+  :: (HasCallStack, MonadTest m, MonadIO m, H.MonadAssertion m, MonadCatch m)
+  => H.ExecConfig -- ^ Specifies the CLI execution configuration.
+  -> EpochStateView -- ^ Current epoch state view for transaction building. It can be obtained
+                    -- using the 'getEpochStateView' function.
+  -> FilePath -- ^ Path to the node configuration file as returned by 'cardanoTestnetDefault'.
```

Comment:
Not for this PR but all of these separate parameters for the configuration file path, socket file path, subfolder etc is an indication that we should have a type that groups together these inputs. However we must also consider what is available in the other configuration related types e.g `TestnetRuntime`, `CardanoTestnetOptions` etc in order to separate concerns and avoid duplicating fields.

Reply:
Yes, I am really tempted to make a monad for tests that abstracts out all this. I can give it a go after the automated drep PRs are merged (to avoid having to rebase a lot)